apps/extension/src/core/views/PermissionRequest.tsx

```diff
@@ -61,17 +61,15 @@ export function PermissionRequest({
 
 function TransactionPermission({ transaction }: { transaction: Transaction }) {
   const { setSettingsShown } = useNav()
-  const { modelId, setModelId } = useModel()
-  const [label, setLabel] = useState("")
+  const { config, setConfig } = useConfig()
   const { object, setObject } = originManager.useObject(transaction.origin.id)
   const requestedModel = transaction.model
 
   useEffect(() => {
     async function checkConfig() {
-      const config = await configManager.getWithDefault(requestedModel)
-      setModelId(config.id)
-      setLabel(config.label)
-      if (configManager.isIncomplete(config)) {
+      const config = await configManager.forModelWithDefault(requestedModel)
+      setConfig(config)
+      if (!configManager.isCredentialed(config)) {
```

Comment:
nit - Something tells me that this procedure should be inside the config provider. Perhaps once we start seeing it being used in other place and form a pattern.

Reply:
ah, including the call to `setSettingsShown`? so the config provider would depend on `useNav`?

It's possible that that could result in some surprising behavior, e.g. someone trying to manage a config that isn't supposed to be the default from some other component. but yeah can think more about it

Reply:
Just the procedure involving `configManager` - it can return the `isCredentialed` as the output maybe :-?...